### PR TITLE
Conform Never to Error, Equatable, Comparable, and Hashable

### PR DESCRIPTION
**Pull request by mdiep**
Implementation for https://github.com/apple/swift-evolution/pull/855.

### CODE REVIEW

File: stdlib/public/core/Policy.swift [hunk 1]
```diff
@@ -28,6 +28,18 @@
 @_frozen
 public enum Never {}
 
+extension Never: Equatable {
+  public static func == (lhs: Never, rhs: Never) -> Bool {
+    switch (lhs, rhs) {
```
Comment (CodaFi) [inline stdlib/public/core/Policy.swift]: This is cute but I would really rather this be spelled out.

Comment (mdiep) [reply stdlib/public/core/Policy.swift]: I’m not sure what you mean. 😕

Comment (stephencelis) [reply stdlib/public/core/Policy.swift]: The switch is required by the compiler right now, no?

Comment (CodaFi) [reply stdlib/public/core/Policy.swift]: `return true` rather than the switch.

Comment (stephencelis) [reply stdlib/public/core/Policy.swift]: Oh, ha. Even did that here: https://github.com/pointfreeco/swift-prelude/blob/master/Sources/Prelude/Equatable.swift#L3 I suppose the switch is only required for `absurd`.

Comment (mdiep) [reply stdlib/public/core/Policy.swift]: I think `switch` is the better option.

`return true` is a bit misleading. It's easy to continue thinking that this code could possibly execute. (`fatalError("This is impossible. A Never value cannot be constructed")` would be better in this regard.)

`switch` educates the reader about `Never` and about `switch`. I think this works as a form of progressive disclosure. You don't need to know that this how it's implemented, but if you go looking, most people are bound to learn something.

I think the swift-evolution discussion bears this out. Several people expressed confusion initially, but walked away with a better understanding.

**Comment by tkremenek** [comment]
@swift-ci build toolchain

**Comment by shahmishal** [comment]
Updated swift-ci comment with correct download links.

**Comment by shahmishal** [comment]
@swift-ci build toolchain

**Comment by tkremenek** [comment]
@swift-ci test

**Comment by tkremenek** [comment]
@mdiep: This proposal has been accepted.  Per the [acceptance rationale](https://forums.swift.org/t/se-0215-conform-never-to-equatable-and-hashable/13586/45) please also add conformances to `Comparable` and `Error`.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Review by lorentey — APPROVED** [review]
Looks good! 👍

**Comment by airspeedswift** [comment]
Can we add some form of test? The usual `checkEquatable` kind of tests won't work because you can't construct one, but forming some code along the lines of the evolution proposal motivation and checking it compiles would be good.

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by mdiep** [comment]
Can this be merged? 🙂

**Comment by mdiep** [comment]
Could this be included in 4.2? Or would that pose a potential source compatibility problem?

(I'm happy to open a PR. I just don't want to waste any effort.)

**Comment by airspeedswift** [comment]
Probably not at this stage. We’re trying to keep it at just bug fixes.

**Pull request merged**